src/components/menu/menu.test.tsx

```diff
@@ -0,0 +1,221 @@
+      <MenuItem href="#">test two</MenuItem>
+      <MenuItem href="#">test three</MenuItem>
+      <MenuItem href="#">test four</MenuItem>
+    </Menu>
+  );
+  const items = screen.getAllByRole("link");
+  items[3].focus();
+  await user.tab({ shift: true });
+
+  expect(items[2]).toBeFocused();
+  await user.tab({ shift: true });
+  expect(items[1]).toBeFocused();
+  await user.tab({ shift: true });
+  expect(items[0]).toBeFocused();
+  await user.tab({ shift: true });
+  expect(document.body).toBeFocused();
+});
+
+test("should not focus the next item in sequence when 'arrowright' key is pressed by user", async () => {
+  const user = userEvent.setup();
+  render(
+    <Menu>
+      <MenuItem href="#">test one</MenuItem>
+      <MenuItem href="#">test two</MenuItem>
+      <MenuItem href="#">test three</MenuItem>
+      <MenuItem href="#">test four</MenuItem>
+    </Menu>
+  );
+  const items = screen.getAllByRole("link");
+  items[0].focus();
+  await user.keyboard("{arrowright}");
+
+  expect(items[0]).toBeFocused();
+  await user.keyboard("{arrowright}");
+  expect(items[0]).toBeFocused();
+  await user.keyboard("{arrowright}");
+  expect(items[0]).toBeFocused();
+});
+
+test("should not focus the previous item in sequence when 'arrowleft' key is pressed by user", async () => {
+  const user = userEvent.setup();
+  render(
+    <Menu>
+      <MenuItem href="#">test one</MenuItem>
+      <MenuItem href="#">test two</MenuItem>
+      <MenuItem href="#">test three</MenuItem>
+      <MenuItem href="#">test four</MenuItem>
+    </Menu>
+  );
+  const items = screen.getAllByRole("link");
+  items[3].focus();
+  await user.keyboard("{arrowleft}");
+
+  expect(items[3]).toBeFocused();
+  await user.keyboard("{arrowleft}");
+  expect(items[3]).toBeFocused();
+  await user.keyboard("{arrowleft}");
+  expect(items[3]).toBeFocused();
+});
+
+test("should not throw an error when conditionally rendered `children` are passed", () => {
+  expect(() => {
+    render(
+      <Menu>
+        {true && <MenuItem href="#">test one</MenuItem>}
+        {false && <MenuItem href="#">test two</MenuItem>}
+      </Menu>
+    );
+  }).not.toThrow();
+});
+
+test("should close any open submenus when a new submenu is opened", async () => {
+  const user = userEvent.setup();
+  render(
+    <Menu>
+      <MenuItem href="#">menu item</MenuItem>
+      <MenuItem submenu="submenu 1">
+        <MenuItem href="#">submenu 1 item 1</MenuItem>
+        <MenuItem href="#">submenu 1 item 2</MenuItem>
+      </MenuItem>
+      <MenuItem submenu="submenu 2">
+        <MenuItem href="#">submenu 2 item 1</MenuItem>
+        <MenuItem href="#">submenu 2 item 2</MenuItem>
+      </MenuItem>
+    </Menu>
+  );
+  const firstSubmenuItem = screen.getByRole("button", { name: "submenu 1" });
+  const secondSubmenuItem = screen.getByRole("button", { name: "submenu 2" });
+  await waitFor(async () => {
+    await user.click(firstSubmenuItem);
+  });
```

Comment:
I do agree that some of our components could do with some further digging, especially where we see significant slowdown in test runs (although I'm less concerned about unit tests slowdown than PW ones). However, I'm reluctant to willingly add extra bloat into the console. The docs suggest to wrap assertions in `waitFor`, but I've not found anything online that suggests my approach is invalid. We decided to use `waitFor` instead of `act` so we will need to wrap the event triggers in `waitFor` to remove the additional warnings.